src/BackendStatusBanner/index.jsx

```diff
@@ -1,39 +1,41 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
 import statusMap from './statusMap.json';
+import { pingStudio } from '../data/actions/pingStudio';
 
 class BackendStatusBanner extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      apiConnectionStatus: 200,
-    };
-  }
-
   componentDidMount() {
-    fetch('/api/assets/course-v1:edX+DemoX+Demo_Course/?page=0&page_size=50&sort=sort&asset_type=', {
-      credentials: 'same-origin',
-      headers: {
-        Accept: 'application/json',
-      },
-    })
-      .then((response) => {
-        if (response.status !== this.state.apiConnectionStatus) {
-          this.setState({
-            apiConnectionStatus: response.status,
-          });
-        }
-      });
+    this.props.pingStudio();
   }
 
   render() {
-    return (this.apiConnectionStatus === 200) ?
+    return (this.props.connectionStatus === 200) ?
       null :
       (
         <div className="api-error">
-          {statusMap[this.state.apiConnectionStatus]}
+          {statusMap[this.props.connectionStatus]}
         </div>
       );
   }
 }
 
-export default BackendStatusBanner;
+BackendStatusBanner.propTypes = {
```

Comment:
`propTypes` is a neat module that enables us to do type checking on the props (props are basically arguments) passed into the component. https://github.com/facebook/prop-types

These props can be passed where the component gets invoked (e.g. `<BackendStatusBanner connectionStatus={200} />, or they can be indirectly passed through Redux, as we are doing below.